scripts/operations/configuration/apply_csm_configuration.sh

```diff
@@ -23,33 +23,10 @@
 # OTHER DEALINGS IN THE SOFTWARE.
 #
 
-err()
-{
-    echo "ERROR: $*" 1>&2
-}
-
-err_exit()
-{
-    err "$@"
-    exit 1
-}
-
-run_cmd()
-{
-    "$@" || err_exit "Command failed with rc $?: $*"
-}
-
-run_mktemp()
-{
-    tmpfile=$(mktemp "$@") || err_exit "Command failed with rc $?: mktemp $*"
-    [[ -n "${tmpfile}" ]] || err_exit "mktemp command passed but gave no output"
-    [[ -e "${tmpfile}" ]] || err_exit "mktemp command passed but '${tmpfile}' does not exist"
-    if [[ $# -gt 0 && "$1" == "-d" ]]; then
-        [[ -d "${tmpfile}" ]] || err_exit "mktemp -d command passed and '${tmpfile}' exists, but is not a directory"
-    else
-        [[ -f "${tmpfile}" ]] || err_exit "mktemp command passed and '${tmpfile}' exists, but is not a regular file"
-    fi
-}
+locOfScript=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
```

Comment:
I feel like we could probably debate this but personally I think using `cd` and `pwd` is sort of overkill in this case. Is it just to check that the directory exists?

Reply:
I copied this from another script in the docs repo, but I believe it is done to avoid getting a relative path to the file. I figured I'd stay consistent with how the other scripts were doing it, if for no other reason than to avoid having all of the scripts look a little different. I don't know if there is an edge case in there that they are trying to avoid by this mechanism.

Reply:
Good with me, didn't realize this was also how other scripts were doing it.

Reply:
Yeah, the ones in upgrade/scripts all do it that way.

It looks to me like it was probably lifted from here: https://stackoverflow.com/a/246128

Reply:
Whoever wrote this should have read the article rather than just copying the top solution 😆